common/thorhelper/thorcommon.cpp

```diff
@@ -2148,6 +2148,9 @@ static bool getTranslators(Owned<const IDynamicTransform> &translator, Owned<con
             if (!translator->canTranslate())
                 throw MakeStringException(0, "Untranslatable record layout mismatch detected for file %s", tracing);
 
+            if (mode == RecordTranslationMode::PayloadRemoveOnly && translator->hasNewFields())
+                throw MakeStringException(0, "Untranslatable record layout mismatch detected for file %s. Expected fields were not found in source.", tracing);
```

Comment:
minor: It is translatable, but mode prohibits it. Consider different message here (and similarly elsewhere).

Reply:
Changed message.